app/models/external_service.rb

```diff
@@ -0,0 +1,27 @@
+# == Schema Information
+#
+# Table name: external_services
+#
+#  id                     :integer          not null, primary key
+#  name                   :string(255)      not null
+#  short_name             :string(255)      not null
+#
+# Indexes
+#
+#  index_external_services_on_short_name  (short_name)
+#
+
+class ExternalService < ActiveRecord::Base
+  has_many :threads, class_name: 'MessageThread', inverse_of: :external_service
+
+  validates :name, presence: true, uniqueness: true
+  validates :short_name, presence: true, uniqueness: true, subdomain: true
```

Comment:
What is this?  As in what is the difference between the name and short name?  Why not just use the name?